server/eclipse-project/src/main/java/us/freeandfair/corla/query/CountyContestResultQueries.java

```diff
@@ -144,17 +144,40 @@ public static CountyContestResult matching(final County the_county,
    * @param the_id The county ID.
    */
   public static Integer deleteForCounty(final Long the_county_id) {
-    final Set<CountyContestResult> results = 
-        forCounty(Persistence.getByID(the_county_id, County.class));
-    Integer n = 0;
-    if (results != null) {
+    final Session s = Persistence.currentSession();
+
+    final Set<CountyContestResult> results =
+      forCounty(Persistence.getByID(the_county_id, County.class));
+
+    final List<Long> contestIds = new ArrayList();
+
+    if (results.size() > 0) {
       for (final CountyContestResult c : results) {
-        Persistence.delete(c);
-        Persistence.delete(c.contest());
-        n++;
+        contestIds.add(c.contest().id());
```

Comment:
`contest()` causes the Contest to be lazy-loaded, so we're still doing that in a loop and that will add significantly to the latency. Hopefully the rest of it is fast enough!